integration/service/pom.xml

```diff
@@ -0,0 +1,209 @@
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-api</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-log4j12</artifactId>
+    </dependency>
+  </dependencies>
+
+  <build>
+    <resources>
+      <resource>
+        <directory>src/main/resources</directory>
+        <filtering>true</filtering>
+      </resource>
+    </resources>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-war-plugin</artifactId>
+        <configuration>
+          <warName>beadledom-integration-service</warName>
+          <attachClasses>true</attachClasses>
+          <failOnMissingWebXml>false</failOnMissingWebXml>
+        </configuration>
+      </plugin>
+      <plugin>
+        <groupId>org.eclipse.jetty</groupId>
+        <artifactId>jetty-maven-plugin</artifactId>
+        <dependencies>
+          <dependency>
+            <groupId>org.eclipse.jetty</groupId>
+            <artifactId>jetty-server</artifactId>
+            <version>${jetty.version}</version>
+          </dependency>
+          <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+            <version>${slf4j.version}</version>
+          </dependency>
+          <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>jul-to-slf4j</artifactId>
+            <version>${slf4j.version}</version>
+          </dependency>
+        </dependencies>
+        <configuration>
+          <jettyXml>${project.basedir}/local/jetty.xml</jettyXml>
+          <webApp>
+            <contextPath>/beadledom-integration-service</contextPath>
+            <descriptor>${project.basedir}/src/main/webapp/WEB-INF/web.xml</descriptor>
+          </webApp>
+          <systemProperties>
+            <systemProperty>
+              <name>PROJECT_VERSION</name>
+              <value>${project.version}</value>
+            </systemProperty>
+            <systemProperty>
+              <name>LOCAL_DIR</name>
+              <value>${project.basedir}/local</value>
+            </systemProperty>
+            <systemProperty>
+              <name>jetty.base</name>
+              <value>${project.build.directory}</value>
+            </systemProperty>
+          </systemProperties>
+        </configuration>
+      </plugin>
+      <plugin>
+        <groupId>io.fabric8</groupId>
+        <artifactId>docker-maven-plugin</artifactId>
+        <version>0.30.0</version>
+        <configuration>
+          <images>
+            <image>
+              <alias>beadledom-integration-service</alias>
+              <name>${project.groupId}/${project.artifactId}:${project.version}</name>
+
+              <build>
+                <cleanup>remove</cleanup>
+                <env>
+                  <PROJECT_VERSION>${project.version}</PROJECT_VERSION>
+                </env>
+                <ports>
+                  <port>8080</port>
```

Comment:
Can the port be randomized? Is there any disadvantage to keeping it just 8080?

Reply:
I'm not sure it could be randomized, I use this port to redirect the traffic through nginx. I don't think I can use environment variables inside of the nginx.config files so it might have to be hardcoded. There could be a disadvantage in hardcoding the port because it makes the tests rather brittle (ie. if any other is running on port 8080 it would fail). I'll can look into using an environment variable.